contracts/stake/src/tests/distribution.rs

```diff
@@ -976,8 +1069,86 @@ fn panic_when_adding_same_distribution_twice() {
     let lp_token = deploy_token_contract(&env, &admin);
     let reward_token = deploy_token_contract(&env, &admin);
 
-    let staking = deploy_staking_contract(&env, admin.clone(), &lp_token.address, &manager, &owner);
+    let staking = deploy_staking_contract(
+        &env,
+        admin.clone(),
+        &lp_token.address,
+        &manager,
+        &owner,
+        &50u32,
+    );
+
+    staking.create_distribution_flow(&manager, &reward_token.address);
+    staking.create_distribution_flow(&manager, &reward_token.address);
+}
+
+#[should_panic(expected = "Stake: Fund distribution: Curve complexity validation failed")]
+#[test]
+fn panic_when_funding_distribution_above_end_distribution_timestamp() {
+    const DISTRIBUTION_MAX_COMPLEXITY: u32 = 1;
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let manager = Address::generate(&env);
+    let owner = Address::generate(&env);
+    let lp_token = deploy_token_contract(&env, &admin);
+    let reward_token = deploy_token_contract(&env, &admin);
+
+    let staking = deploy_staking_contract(
+        &env,
+        admin.clone(),
+        &lp_token.address,
+        &manager,
+        &owner,
+        &DISTRIBUTION_MAX_COMPLEXITY,
+    );
 
     staking.create_distribution_flow(&manager, &reward_token.address);
+
+    reward_token.mint(&admin, &2000);
+
+    staking.fund_distribution(&admin, &0, &600, &reward_token.address, &1000);
+    staking.fund_distribution(&admin, &0, &600, &reward_token.address, &1000);
```

Comment:
Make complexity 2.
Create two distribution flows succesfully (don't create the identical one twice...)
Then 3rd should fail one way or another.

Reply:
using complexity 1 doesn't work, it's why I previously had it with complexity 1 and used 2 fund_distributions to make it fail